### PR TITLE
[flashing] Support the check for product on Windows. Fixes MER#1961

### PR DESCRIPTION
**Pull request by Thaodan**
Modifers @DEVICES@ to support matching on supported devices and
changes the layout of the comment containing all supported devices as in flash.sh

### CODE REVIEW

**Comment by mkosola** [comment]
At least this does not work for me... I get following error:
`Searching a compatible device...
GOTO was unexpected at this time.`

Also product is now in two places on windows flashing script

**Comment by Thaodan** [comment]
I updated the pr to the current changes but need to add a second string to match devices found.
However I still need to test this later.

File: kickstart/pack/h4113/hybris [hunk 1]
```diff
@@ -142,21 +142,27 @@ FILES_TO_COPY="*.urls"
 if [ "@DEVICEMODEL@" = "h3113" ]; then
   sed -e "s/\@DEVICES\@/\-e \"H3113\" -e \"H3123\" -e \"H3133\"/g" -i flash.sh
   sed -e "s/\@DEVICES\@/:: H3113\r\n:: H3123\r\n:: H3133/g" -i flash-on-windows.bat
+  sed -e "s/\@DEVICES_FINDSTR\@/H3123 H3133/g" -i flash-on-windows.bat
```
Comment (mlehtima) [inline kickstart/pack/h4113/hybris]: Is there any reason to add a second sed command for windows script? You could just replace the existing one as that is not really used for anything just to add a comment to the script.

File: sparse/boot/flash-on-windows.bat [hunk 2]
```diff
@@ -58,33 +58,40 @@ exit /b 1
 
 :no_error_serialnumbers
 
+for %%d in ( %serialnumbers% ) do (
+  set fastbootcmd=%fastbootcmd_no_devic% -s %%d
```
Comment (mlehtima) [inline sparse/boot/flash-on-windows.bat]: Typo, should have fastbootcmd_no_device

**Comment by mlehtima** [comment]
Wondering if we need to support having multiple devices connected? That just makes the script somewhat more complicated and it's quite rare use case to have multiple devices of different type in fastboot mode connected to the computer.

File: sparse/boot/flash-on-windows.bat (hunk 2 above)
Comment (Thaodan) [reply sparse/boot/flash-on-windows.bat]: Fixed.

File: kickstart/pack/h4113/hybris (hunk 1 above)
Comment (Thaodan) [reply kickstart/pack/h4113/hybris]: Thanks for the suggestion. I've changed the layout of the which contained @DEVICES@ to do this.

**Comment by jovirkku** [comment]
At least for now, our requirement has been to also support Windows 7 and Windows 8. Windows 7 does not have PowerShell (not sure about Win 8). Furthermore, there has been some cases where a Windows PC has contained some PowerShell environment variables (defined on that PC) which have caused conflicts with the batch script. So we shouldn't rush into PowerShell before considering the consequences.

**Comment by mkosola** [comment]
We had to modify your code a bit and we did new pull request. Fix is here https://github.com/mer-hybris/droid-config-sony-nile/pull/114

**Pull request closed without merging**